src/libcore/option.rs

```diff
@@ -10,9 +10,9 @@
 
 //! Optional values
 //!
-//! Type `Option` represents an optional value: every `Option`
+//! Type `Option<T>` represents an optional value: every `Option<T>`
```

Comment:
Is that going to be the new convention? It seems way too redundant and a bit distracting. Is `HashMap::contains_key`'s documentation going to become

```
Returns true if the `HashMap<K, V, S>` contains a value for the specified key.
```

My preference is to use a lowercase non-Rust-type word when the meaning is obvious, which it is when we're talking about methods on a type, but I guess this needs some kind of guidelines RFC. At the very least, I don't think we need to use the type parameter everywhere.